From: =?utf-8?q?IOhannes_m_zm=C3=B6lnig?= <umlaeute@debian.org>
Date: Tue, 29 Jan 2019 11:31:36 +0100
Subject: remove C-macros for timestamps

Last-Update: 2015-05-20

For reproducible builds, we don't want timestamps in the builds.
Instead we should set the version to be displayed to the Debian
package version (use PD_TEST_VERSION) for this.
Last-Update: 2015-05-20
---
 src/s_main.c | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

--- puredata.orig/src/s_main.c
+++ puredata/src/s_main.c
@@ -32,8 +32,12 @@
 stringify(PD_MINOR_VERSION) "." stringify(PD_BUGFIX_VERSION) "\
  (" stringify(PD_TEST_VERSION) ")";
 
-char pd_compiletime[] = __TIME__;
-char pd_compiledate[] = __DATE__;
+#ifdef PD_TIMESTAMP_STRING
+char pd_compiletimestamp[] = PD_TIMESTAMP_STRING;
+#else
+char pd_compiletimestamp[] = "compiled " __TIME__ " " __DATE__;
+#endif
+
 
 void pd_init(void);
 int sys_argparse(int argc, char **argv);
@@ -375,8 +379,8 @@
     if (sys_argparse(argc-1, argv+1))           /* parse cmd line args */
         return (1);
     sys_afterargparse();                    /* post-argparse settings */
-    if (sys_verbose || sys_version) fprintf(stderr, "%s compiled %s %s\n",
-        pd_version, pd_compiletime, pd_compiledate);
+    if (sys_verbose || sys_version) fprintf(stderr, "%s %s\n",
+        pd_version, pd_compiletimestamp);
     if (sys_verbose)
         fprintf(stderr, "float precision = %lu bits\n", sizeof(t_float)*8);
     if (sys_version)    /* if we were just asked our version, exit here. */
